src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -1,50 +1,60 @@
 package seedu.address.logic.commands;
 
 import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MODULE;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
 
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
 import java.util.stream.Collectors;
 
 import seedu.address.commons.util.ToStringBuilder;
 import seedu.address.logic.Messages;
 import seedu.address.model.Model;
+import seedu.address.model.person.ModuleRoleContainsKeywordsPredicate;
 import seedu.address.model.person.NameContainsKeywordsPredicate;
+import seedu.address.model.person.Person;
 
 
 /**
- * Finds and lists all persons in address book whose name contains any of the argument keywords.
+ * Finds and lists all persons in address book whose name or module contains any of the argument keywords.
  * Keyword matching is case insensitive.
  */
 public class FindCommand extends Command {
 
     public static final String COMMAND_WORD = "find";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose names contain any of "
-            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds persons by their names, module-role pairs"
+            + "(or both), matching any (combination) of the specified keywords (case-insensitive) "
+            + "and displays them as a list with index numbers.\n"
             + "Parameters: "
-            + PREFIX_NAME + "KEYWORD ["
-            + PREFIX_NAME + "MORE_KEYWORDS]...\n"
+            + "(" + PREFIX_NAME + "KEYWORD | " + PREFIX_MODULE + "KEYWORD)+\n"
             + "Example: " + COMMAND_WORD + " "
             + PREFIX_NAME + "alice "
             + PREFIX_NAME + "bob "
-            + PREFIX_NAME + "charlie";
+            + PREFIX_MODULE + "CS1101S "
+            + PREFIX_MODULE + "CS2030S";
 
-    private final NameContainsKeywordsPredicate predicate;
+    private final List<Predicate<Person>> predicates;
 
-    public FindCommand(NameContainsKeywordsPredicate predicate) {
-        this.predicate = predicate;
+    public FindCommand(List<Predicate<Person>> predicates) {
+        this.predicates = predicates;
     }
```

Comment:
from our usage, we expect either 1 or 2 Predicates in the list, so maybe can add an assert statement for defensive programming?

Reply:
(an alternative is to use `Optional`, but I think that would take a bit too much time for refactoring)